controllers/terraform_controller.go

```diff
@@ -586,9 +607,11 @@ terraform {
 
 	}
 
+	justApplied := false
 	outputs := map[string]tfexec.OutputMeta{}
 	if r.shouldApply(terraform) {
 		terraform, err = r.apply(ctx, terraform, tf, revision, &outputs)
+		justApplied = true
```

Comment:
This `justApplied` check is not enough here.
Please check that the Apply condition is "Success" before doing health checks.